### ISSUE TITLE
1.19.3 doesnt work on 1.19.3

### ISSUE THREAD

**Issue by krjencik** [YeetmanLord/So-Many-Enchants#19]
**Describe the bug**
i downloaded this mod on 1.19.3. Minecraft then crashed right on startup.
**To Reproduce**

**Expected behavior**
be able to play
**Screenshots**

**Additional context**
[latest.log](https://github.com/YeetmanLord/So-Many-Enchants/files/10324059/latest.log)
[debug.log](https://github.com/YeetmanLord/So-Many-Enchants/files/10324060/debug.log)

**Comment by Coraline774477** [comment]
I've encountered the same thing and my latest log is similar to yours krjencik.

krjencik's: _[30Dec2022 10:50:51.433] [main/ERROR] [net.minecraftforge.fml.loading.ModSorter/LOADING]: Missing or unsupported mandatory dependencies:
	Mod ID: 'forge', Requested by: 'so_many_enchants', Expected range: '[43,44]', Actual version: '44.0.41'_


Mine: _[11:41:44] [main/ERROR]: Missing or unsupported mandatory dependencies:
	Mod ID: 'forge', Requested by: 'so_many_enchants', Expected range: '[43,44]', Actual version: '44.1.4'_

That's what's in my latest log as well, also this is what's at the bottom of my latest log:

_[11:41:49] [main/WARN]: Invalid registry value type detected for PerfOS counters. Should be REG_DWORD. Ignoring: HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\PerfOS\Performance\Disable Performance Counters._

Somethin tells me that you @YeetmanLord need to update the mod's compatibility above 1.19.2 (43.2.2) to 1.19.3 (44.1.x) since Lex and his crew have done some breaking changes as well as _[43,44]_ isn't specifying any specific version of forge to use for So Many Enchants 1.19.3 which is possibly causing forge as well as Minecraft to get confused as to which version this mod is  made for.

Since Lex and his crew have made some game breaking changes possibly you may need to rewrite some of the mod especislly when Mojang had changed the rendering engine again as of changing to a brand new one for optimal performance possibly.

In conclusion, So Many Enchants is NOT compatible with Minecraft 1.19.3 and Forge 1.19.3-44.1.x (I'm currently on 44.1.4 btw)!

Which sucks cause I really was lookin forward to So Many Enchants 1.19.3 when it's not even compatible with Forge 44.1.x (44.1.0-44.1.5).

What's strange is that the version checker didn't flag krjencik and I using the wrong forge version, especially since we don't know what [43,44] is, change that to something that both us humans and forge to understand cause that doesn't specify what forge version So Many Enchants 1.19.3 is really for which is why the usual screen that pops up when we have mix match forge versions for what the mod was really made for didn't pop up and instead just insta crashed us on startup!..

**Comment by max-widmaier** [comment]
Ok, I'll try to fix it soon. Thanks for the notice.

**Comment by Coraline774477** [comment]
I had forgotten to mention that most of the game breaking changes were in the rendering and GUI department which SME doesn't really have anything to do with GUIs and rendering but that doesn't mean that you MIGHT want to rewrite some of the mod but not the whole thing. BUT you definitely will have to change the versioning so that both Minecraft and Forge would understand what version the 1.19.3 version of SME's made for. Meaning that you'll need to re-upload the 1.19.3 version as an actual port instead of tryin to just port it over without actually changing anything, especially the forge versioning within SME's code.

BTW just letting ya know that I've decided to go back to 1.19.2 and even 1.18.2 (Tinkers' Construct 3) cause there's still some mods that haven't been ported over to *1.19.3+ yet like Modern UI for an example which there's an obvious reason why that mod hasn't been ported over to 1.19.3 yet. Also currently how it is without most of those mods makes me feel like there's something's missing from my modding experience, especially with Player Companions mod since the fairy in that mod really is a life saver in certain circumstances/situations
(except for the times while battling against creepers and any bosses like the Wither, the Ender Dragon and any modded bosses that has the potential to one shot the player even with enchanted diamond/netherite armor without any modded armors added into the game.
Why I mentioned the creeper is cause if he's standing right next to you with your fairy/fairies that he'll one shot her/them with his powerful explosion while possibly one shooting you in the process,  especially the super charged creepers if you're not fully decked out in enchanted diamond/netherite armor).

*why I say 1.19.3+ is cause there's snapshots for 1.19.4 already.

One more thing, another reason why I've decided to go back to 1.19.2-1.18.2 is due to performance BUT with diminished amount of mods that I play with since my war against my own computer in the RAM development is really getting to me now.
So instead of say 70+ mods now I've dwindled down to between 45-55 mods due to me running out of memory both physically and virtually. Who knew that Minecraft was more RAM intensive than CPU and GPU?..
Soon I'll be stuck in vanilla minecraft which if this war against the RAM gets even worse then I'll be forced to quit playing minecraft till I could get a MUCH better pc than a damn "gaming" laptop or even laptops at all since laptops clearly aren't really meant for real deal video gaming such as AAA games like COD an even Minecraft, especially modded...

**Comment by max-widmaier** [comment]
Ya, I have to do some special rendering stuff for enchantable blocks (specifically, chests and shulker boxes), so I see why it broke. I'll probably just hold off on 1.19.3 for a bit.

**Comment by Coraline774477** [comment]
Btw would you mind if we dm or you have  discord server? So that we wouldn't need to have a conversation on Github in an issue report (especially someone else's issue report)?
Also I dont mind if you laid off 1.19.3 for now since I did say that I'm personally gonna be on 1.18.2/1.19.2 for some time as well as for some performance reasons.
While I'm waiting for those other mods to finish with their port as well as SME's real 1.19.3 port which its first port to 1.19.3 was obviously just a test for us to test to see if it'll work which it obviously didn't. 

Also why I didn't really mention that SME has any rendering is cause I don't really notice that the mod does have some rendering in it which mainly is why the mod didn't work.

**Comment by max-widmaier** [comment]
I can get that set up
